### DISCUSSION TITLE
If the rendertarget texture can transfer raw value to shader ？

### DISCUSSION THREAD

**Discussion by fanbinnet** [google/filament#4998] [Q&A]
I use an framebuffer texture to compute the particles position,then transfer the texture to next compute pass.
The position range is (-50.0 ,50.0). 

In next compute pass's shader ,I notice the pos value is in an range (0.0,1.0).
float4 pos = texture(materialParams_posTexture, uv);

How Can I get the raw value ?

Thanks!
       
```
 particleFboPos.color = Texture.Builder()
            .width(width).height(height).levels(1)
            .usage(Texture.Usage.COLOR_ATTACHMENT or Texture.Usage.SAMPLEABLE)
            .format(Texture.InternalFormat.RGBA16F)
            .build(engine)
        particleFboPos.target = RenderTarget.Builder()
            .texture(RenderTarget.AttachmentPoint.COLOR, particleFboPos.color)
            .build(engine)

particleFboPos.view.colorGrading = ColorGrading.Builder().quality(ColorGrading.QualityLevel.HIGH)
            .toneMapper(ToneMapper.Linear())
            .build(engine)
```

**Comment by bejado** [comment]
Seems like you'd want to use [`texelFetch`](https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/texelFetch.xhtml) instead to avoid filtering. Have you tried filling the texture entirely with a single value, like 50.0 and seeing what value `texture`/`texelFetch` returns?

**Reply by fanbinnet** [reply]
Thanks for the replay，I'll try the texelFetch and give the feedback.

**Reply by fanbinnet** [reply]
@bejado I have tested texture/texelFetch. The method texture get wrong result data mabye is because of filtering,and only get the ragne (0.0,1.0). The texelFetch can get right result data. But the range is also (0.0,1.0).
I have generate random particles position xyz in （-50.0,50.0）,but only the particles in (0.0,1.0) can be show.

I guess If the ToneMapper.Linear() clamp the data to (0.0,1.0). 
If I set isPostProcessingEnabled false ,I can't get any data in material.

**Reply by bejado** [reply]
Yes, you definitely want to make sure post-processing is disabled. Are you able to upload a complete code sample? You can also look at our [rendertarget](https://github.com/google/filament/blob/main/samples/rendertarget.cpp) sample for reference.

**Reply by fanbinnet** [reply]
When the post-processing is disabled, if the rendertarget can work ?

**Reply by bejado** [reply]
Yes, note that we [disable post-processing](https://github.com/google/filament/blob/7873e6f5aa39f66684f31815ac3c33fd9261f000/samples/rendertarget.cpp#L204) in the rendertarget sample.

**Reply by fanbinnet** [reply]
Thanks，I'll try again and give me my feedback.
